src/routines/solve_routines.jl

```diff
@@ -130,8 +136,140 @@ function run_sim_model!(sim::Simulation; verbose::Bool = false, kwargs...)
             @assert stage.executions == stage.execution_count
             stage.execution_count = 0 # reset stage execution_count
         end
+        
+    end
+    if (:no_dict in keys(kwargs)) == true
+        return
+    else
+        date_run = convert(String,last(split(dirname(sim.ref.raw),"/")))
+        references = make_references(sim, date_run)
+        return references
     end
 
-    return
+end
+""" 
+    make_references(sim::Simulation, date_run::String)
+
+Creates a dictionary of variables with a dictionary of stages
+that contains dataframes of date/step/and desired file path
+so that the results can be parsed sequentially by variable
+and stage type.
+
+**Note:** make_references can only be run after run_sim_model
+or else, the folder structure will not yet be populated with results
+
+# Arguments
+-`sim::Simulation = sim`: simulation object created by Simulation()
+-`date_run::String = "2019-10-03T09-18-00-test"``: the name of the file created
+that contains the specific simulation run of the date run and "-test"
+
+# Example
+```julia
+sim = Simulation("test", 7, stages, "/Users/lhanig/Downloads/"; 
+verbose = true, system_to_file = false)
+run_sim_model!(sim::Simulation; verbose::Bool = false, kwargs...)
+references = make_references(sim, "2019-10-03T09-18-00-test")
+```
+"""
+
+function make_references(sim::Simulation, date_run::String)
```

Comment:
Any reason to put this in the solve routines?

Reply:
hmmm not particularly, should I just make a new file?

Reply:
What is the `add_reference` function supposed to do?